### PR TITLE
Spamcop IP reputation

### PR DESCRIPTION
**Pull request by Hruuttila**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
Integration to fetch results from SpamCop.net for IP reputation.

## Minimum version of Demisto
- [ ] 5.0.0
- [x] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [x] Documentation

### CODE REVIEW

**Comment by reutshal** [comment]
Hi @Hruuttila ! Thanks for contributing.
We have two options here:
* If you want this to be a community contribution, please change the support level in the pack metadata and I'll review the pack as a community contribution.
* If you want it to be xsoar supported, we will have to open an issue and then it will be decided when to take it.

**Comment by Hruuttila** [comment]
Hey @reutshal , changed support to community.

File: Packs/Spamcop/Integrations/Spamcop/Spamcop.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    arguments:
+    - name: ip
+      required: true
+      description: The IP which details you want to find
+      default: true
+      isArray: true
+    outputs:
+    - contextPath: Spamcop.IP
+      description: IP details
+      type: String
```
Comment (reutshal) [inline Packs/Spamcop/Integrations/Spamcop/Spamcop.yml:24]: Please add here also the dbotscore paths + IP paths, for example:
DBotScore.Indicator, IP.Address, etc..
You can find all relevant keys and descriptions here And also add these as command outputs. You can find the standards here: https://xsoar.pan.dev/docs/integrations/context-standards

File: Packs/Spamcop/Integrations/Spamcop/Spamcop.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+def results_return(command, thingtoreturn):
+    ip_reputation = {
+        'indicator': thingtoreturn['Address'],
+    }
+    try:
+        if thingtoreturn['Malicious']['Vendor']:
+            score = Common.DBotScore.BAD
+    except LookupError:
+        score = Common.DBotScore.NONE
+    dbot_score = Common.DBotScore(
```
Comment (reutshal) [inline Packs/Spamcop/Integrations/Spamcop/Spamcop.py:19]: You also use the readable_output entry and return some message to the user

**Review by reutshal — COMMENTED** [review]
Thanks for contributing!
Overall looks good, see my minor comments.

**Comment by reutshal** [comment]
@Hruuttila Please share your availability for a short demo, I'm free next week from Tuesday.

File: Packs/Spamcop/Integrations/Spamcop/Spamcop.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    except Exception:
+        return 'ok'
+
+
+def main():
+    """
+        PARSE AND VALIDATE INTEGRATION PARAMS
+    """
+
+    LOG(f'Command being called is {demisto.command()}')
```
[the remark this answers is not in the document]
Comment (Hruuttila) [reply Packs/Spamcop/Integrations/Spamcop/Spamcop.py]: This is coming from the PyCharm template directly. I hope that gets updated with these corrections also to reflect the new way of doing things.

File: Packs/Spamcop/Integrations/Spamcop/Spamcop.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    Returning 'ok' indicates that the integration works like it is supposed to. Connection to the service is successful.
+    Returns:
+        'ok' if test passed, anything else will fail the test.
+    """
+    result = socket.gethostbyname('www.spamcop.net')
+    reverselist = str(result).split('.')
+    address = reverselist[3] + '.' + reverselist[2] + '.' + reverselist[1] + '.' + reverselist[0] + '.bl.spamcop.net'
+    try:
+        testresult = socket.gethostbyname(address)
+        return 'Test Failed. Spamcop is blocklisted ' + str(testresult)
```
[the remark this answers is not in the document]
Comment (Hruuttila) [reply Packs/Spamcop/Integrations/Spamcop/Spamcop.py:64]: Yeah, that is how Spamcop service works. You do a reverse DNS query and if it replies with not found then its not blacklisted. In this case I am checking the www.spamcop.net as a test case and figured out that they most likely will not blacklist their own domain. So if the resolution succeeds -> Return error, if it fails create exception and return ok as then its not blacklisted.

File: Packs/Spamcop/Integrations/Spamcop/Spamcop.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        outputs_prefix='Spamcop.' + str(command),
+        outputs_key_field='indicator',
+        outputs=ip_reputation,
+        indicators=[ip]
+    )
+    return_results(results)
+
+
+def get_ip_details(ip):
+    reverselist = str(ip).split('.')
```
[the remark this answers is not in the document]
Comment (Hruuttila) [reply Packs/Spamcop/Integrations/Spamcop/Spamcop.py:39]: Done

**Comment by Hruuttila** [comment]
@reutshal How about review on Tuesday at 10 am Israel time?

File: Packs/Spamcop/Integrations/Spamcop/Spamcop.yml (hunk 1 above)
Comment (Hruuttila) [reply Packs/Spamcop/Integrations/Spamcop/Spamcop.yml:24]: done

File: Packs/Spamcop/Integrations/Spamcop/Spamcop.py [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    address = reverselist[3] + '.' + reverselist[2] + '.' + reverselist[1] + '.' + reverselist[0] + '.bl.spamcop.net'
+    try:
+        result = socket.gethostbyname(address)
+        if result == '127.0.0.2':
+            data = {'Address': ip,
+                    'Malicious': {'Vendor': 'Spamcop',
+                                  'Description': 'IP was found to be on the Spamcop block list'}}
+            return data
+    except Exception:
```
[the remark this answers is not in the document]
Comment (Hruuttila) [reply Packs/Spamcop/Integrations/Spamcop/Spamcop.py:48]: Done. Added exception handling for just the one I am looking for and then others to cover error cases.

File: Packs/Spamcop/Integrations/Spamcop/Spamcop.py (hunk 2 above)
Comment (Hruuttila) [reply Packs/Spamcop/Integrations/Spamcop/Spamcop.py:19]: lets leave some improvement for v2 ;)

File: Packs/Spamcop/Integrations/Spamcop/Spamcop.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (reutshal) [reply Packs/Spamcop/Integrations/Spamcop/Spamcop.py:87]: I will open an issue for this. thanks

File: Packs/Spamcop/Integrations/Spamcop/Spamcop.py [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+    except exception as e:
+        return f'Error, {e.message}. With Args {e.args}.'
+
+
+def main():
+    """
+        PARSE AND VALIDATE INTEGRATION PARAMS
+    """
+
+    LOG(f'Command being called is {demisto.command()}')
```
Comment (reutshal) [inline Packs/Spamcop/Integrations/Spamcop/Spamcop.py:86]: Why did you revert the change?

Comment (Hruuttila) [reply Packs/Spamcop/Integrations/Spamcop/Spamcop.py:86]: by mistake. will change it back.

**Review by reutshal — APPROVED** [review]
